presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -139,7 +142,18 @@ public void start(HiveSplitSource splitSource)
     {
         this.hiveSplitSource = splitSource;
         for (int i = 0; i < maxPartitionBatchSize; i++) {
-            ResumableTasks.submit(executor, new HiveSplitLoaderTask());
+            UserGroupInformation currentHdfsUser = getCurrentHdfsUser();
+            ResumableTasks.submit(executor, new HiveSplitLoaderTask(currentHdfsUser));
+        }
+    }
+
+    private UserGroupInformation getCurrentHdfsUser()
+    {
+        try {
+            return UserGroupInformation.getCurrentUser();
+        }
+        catch (IOException e) {
+            throw Throwables.propagate(e);
```

Comment:
IOException could be thrown only for `tocken` authentication from `Credentials.readTokenStorageFile`. As we are not using that type of authentication i don't think that it make sense to add special exception code for that particular issue.